File: SwiftAlgorithm/BOJ/11399.swift
//
//  main.swift
//  SwiftAlgorithm
//
//  Created by 유정주 on 2022/03/14.
//

//11399 ATM
import Foundation

let n = Int(readLine()!)!
let input = readLine()!.components(separatedBy: " ").map { Int($0)! }.sorted()

var result = 0, sum = 0
for i in 0..<input.count {
    sum += input[i]
    
    result += sum
}

print(result)

File: Python/BOJ/그리디/11399.py
//
//  11399.py
//  SwiftAlgorithm
//
//  Created by 유정주 on 2023/06/05.
//

import sys; readLine = sys.stdin.readline

n = int(readLine())
items = list(map(int, readLine().split()))

items.sort()

result = 0
sum = 0
for item in items:
  sum += item
  result += sum

print(result)
